src/goto-diff/goto_diff_base.cpp

```diff
@@ -11,95 +11,134 @@ Author: Peter Schrammel
+/// Output function information in plain text format
+/// \param function_name: the function id
+/// \param goto_model: the goto model
+void goto_difft::output_function(
+  const irep_idt &function_name,
+  const goto_modelt &goto_model) const
 {
-  // the function may be in goto_model1 or goto_model2
-  if(
-    goto_model1.goto_functions.function_map.find(function_name) !=
-    goto_model1.goto_functions.function_map.end())
+  const auto goto_function_it =
+    goto_model.goto_functions.function_map.find(function_name);
+  PRECONDITION(goto_function_it != goto_model.goto_functions.function_map.end());
+
+  namespacet ns(goto_model.symbol_table);
+  const symbolt &symbol = ns.lookup(function_name);
+  result() << "  " << symbol.location.get_file() << ": " << function_name << "\n";
+  if(options.get_bool_option("show-properties"))
   {
-    const symbolt &symbol =
-      namespacet(goto_model1.symbol_table).lookup(function_name);
-    result["sourceLocation"] = json(symbol.location);
+    const goto_programt &goto_program = goto_function_it->second.body;
+      for(const auto &ins : goto_program.instructions)
+      {
+        if(!ins.is_assert())
+          continue;
+
+        const source_locationt &source_location = ins.source_location;
+        irep_idt property_id = source_location.get_property_id();
+        result() << "    " << property_id << '\n';
+      }
   }
-  else if(
-    goto_model2.goto_functions.function_map.find(function_name) !=
-    goto_model2.goto_functions.function_map.end())
+}
+
+/// Convert a function group to JSON
+/// \param result: the JSON array to be populated
+/// \param function_group: set of function ids in the group
+/// \param goto_model: the goto model
+void goto_difft::convert_function_group_json(
+  json_arrayt &result,
+  const irep_id_sett &function_group,
+  const goto_modelt &goto_model) const
+{
+  for(const auto &function_name : function_group)
   {
-    const symbolt &symbol =
-      namespacet(goto_model2.symbol_table).lookup(function_name);
-    result["sourceLocation"] = json(symbol.location);
+    convert_function_json(
+      result.push_back(jsont()).make_object(), function_name, goto_model);
   }
+}
+
+/// Convert function information to JSON
+/// \param result: the JSON object to be populated
+/// \param function_name: the function id
+/// \param goto_model: the goto model
+void goto_difft::convert_function_json(
+  json_objectt &result,
+  const irep_idt &function_name,
+  const goto_modelt &goto_model) const
+{
+  const auto goto_function_it =
+    goto_model.goto_functions.function_map.find(function_name);
+  PRECONDITION(goto_function_it != goto_model.goto_functions.function_map.end());
+
+  namespacet ns(goto_model.symbol_table);
   result["name"]=json_stringt(id2string(function_name));
+
+  const goto_programt &goto_program = goto_function_it->second.body;
+  const symbolt &symbol = ns.lookup(function_name);
+  result["sourceLocation"] = json(symbol.location);
```

Comment:
The order of instructions in this function seems to be almost random.  Here's what I would do:

````c++
const namespacet ns(goto_model.symbol_table);
const symbolt &symbol = ns.lookup(function_name);

result["name"]=json_stringt(id2string(function_name));
result["sourceLocation"] = json(symbol.location);

if(options.get_bool_option("show-properties"))
{
  const auto goto_function_it = goto_model.goto_functions.function_map.find(function_name);
  ...
````